packages/gatsby-cli/src/reporter/catch-exit-signals.ts

```diff
@@ -5,20 +5,43 @@
 import signalExit from "signal-exit"
 import { getStore } from "./redux"
 import { createPendingActivity } from "./redux/actions"
-import { ActivityStatuses } from "./constants"
+import { ActivityStatuses, ActivityTypes } from "./constants"
 import { reporter } from "./reporter"
 
-const interruptActivities = (): void => {
+const interruptActivities = ({
+  showCancelledPendingActivities = false,
+}: {
+  showCancelledPendingActivities?: boolean
+} = {}): void => {
   const { activities } = getStore().getState().logs
+  const cancelledPendingActivitiesIds: Array<string> = []
   Object.keys(activities).forEach(activityId => {
     const activity = activities[activityId]
     if (
       activity.status === ActivityStatuses.InProgress ||
       activity.status === ActivityStatuses.NotStarted
     ) {
       reporter.completeActivity(activityId, ActivityStatuses.Interrupted)
+      if (activity.type === ActivityTypes.Pending) {
+        // pending activities are hidden from UI, so we construct
+        // list of pending activities to log them to make it easier
+        // to debug any timing issues related to reporting
+        // global status (SET_STATUS)
+        cancelledPendingActivitiesIds.push(`"${activityId}"`)
+      }
     }
   })
+
+  if (
+    showCancelledPendingActivities &&
+    cancelledPendingActivitiesIds.length > 0
+  ) {
+    reporter.warn(
+      `There were pending activities:\n${cancelledPendingActivitiesIds.join(
+        `, `
+      )}.\nPlease, create bug report, if you see this warning when exiting idle process.`
+    )
```

Comment:
I'm not sure how to word properly this. Users can exit when legitimately in progress and the warning then might be confusing.

For the most part this is most helpful in CIs (where SET_STATUS is most often used), so maybe I should enable this logs there only to not confuse users on their local runs?

Reply:
I'd do something like "Please create a bug report if you see this warning and you thought you were existing an idle Gatsby process."